pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/creator/IndexCreationContext.java

```diff
@@ -0,0 +1,467 @@
+    }
+
+    @Override
+    public int getMaxRowLengthInBytes() {
+      return _delegate.getMaxRowLengthInBytes();
+    }
+
+    @Override
+    public boolean isSorted() {
+      return _delegate.isSorted();
+    }
+
+    @Override
+    public int getCardinality() {
+      return _delegate.getCardinality();
+    }
+
+    @Override
+    public int getTotalNumberOfEntries() {
+      return _delegate.getTotalNumberOfEntries();
+    }
+
+    @Override
+    public int getTotalDocs() {
+      return _delegate.getTotalDocs();
+    }
+
+    @Override
+    public boolean hasDictionary() {
+      return _delegate.hasDictionary();
+    }
+  }
+
+  class BloomFilter extends Wrapper {
+
+    private final BloomFilterConfig _bloomFilterConfig;
+
+    public BloomFilter(IndexCreationContext wrapped, BloomFilterConfig bloomFilterConfig) {
+      super(wrapped);
+      _bloomFilterConfig = bloomFilterConfig;
+    }
+
+    @Nullable
+    public BloomFilterConfig getBloomFilterConfig() {
+      return _bloomFilterConfig;
+    }
+  }
+
+  class Forward extends Wrapper {
+
+    private final ChunkCompressionType _chunkCompressionType;
+    private final Map<String, Map<String, String>> _columnProperties;
+
+    Forward(IndexCreationContext delegate,
+        ChunkCompressionType chunkCompressionType,
+        @Nullable Map<String, Map<String, String>> columnProperties) {
+      super(delegate);
+      _chunkCompressionType = chunkCompressionType;
+      _columnProperties = columnProperties;
+    }
+
+    public ChunkCompressionType getChunkCompressionType() {
+      return _chunkCompressionType;
+    }
+
+    @Nullable
+    public Map<String, Map<String, String>> getColumnProperties() {
+      return _columnProperties;
+    }
+  }
+
+  class Geospatial extends Wrapper {
+
+    private final H3IndexConfig _h3IndexConfig;
+
+    Geospatial(IndexCreationContext delegate, H3IndexConfig h3IndexConfig) {
+      super(delegate);
+      _h3IndexConfig = h3IndexConfig;
+    }
+
+    public H3IndexConfig getH3IndexConfig() {
+      return _h3IndexConfig;
+    }
+  }
+
+  class Inverted extends Wrapper {
+
+    Inverted(IndexCreationContext delegate) {
+      super(delegate);
+    }
+  }
+
+  class Json extends Wrapper {
+
+    Json(IndexCreationContext delegate) {
+      super(delegate);
+    }
+  }
+
+  class Range extends Wrapper {
+
+    private final Comparable<?> _min;
+    private final Comparable<?> _max;
+    private final int _rangeIndexVersion;
+
+
+    Range(IndexCreationContext delegate, int rangeIndexVersion, Comparable<?> min, Comparable<?> max) {
+      super(delegate);
+      _rangeIndexVersion = rangeIndexVersion;
+      _min = min;
+      _max = max;
+    }
+
+    public Comparable<?> getMin() {
+      return _min;
+    }
+
+    public Comparable<?> getMax() {
+      return _max;
+    }
+
+    public int getRangeIndexVersion() {
+      return _rangeIndexVersion;
+    }
+  }
+
+  class Text extends Wrapper {
+    private final boolean _commitOnClose;
```

Comment:
This is not really required as it only applies to mutable index (for consuming segment)

Reply:
well, it was hardcoded to be always true, with a comment to name it. Now it has a variable to give it a name and declare its purpose.